data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java

```diff
@@ -115,6 +115,13 @@ void createWithAadAccessTokenAuthentication() {
                 .createWithAadAccessTokenAuthentication("resource.uri", "token"));
     }
 
+    @Test
+    @DisplayName("validate createWithConnectionStringAuthentication  throws IllegalArgumentException exception when missing or invalid parameters")
```

Comment:
Please add more tests for parsing a connection string. What if there are spaces? unexpected order? etc